Automattic/woocommerce-payments
# Update e2e tests for compatibility with WCPay Dev Tools revamp

Fixes #6027 

#### Changes proposed in this Pull Request

Description: Since [the WCPay Dev Tools revamp PR](https://github.com/Automattic/woocommerce-payments-dev-tools/pull/84) has significantly changed the Dev Tools page markup, the parts of our e2e tests that rely on the Dev Tools need updating.

Note: This PR is **to be merged AFTER** the [WCPay Dev Tools PR](https://github.com/Automattic/woocommerce-payments-dev-tools/pull/84) is merged! Until that time, some e2e tests (mainly `upe` and `upe-split`) run via GitHub actions will fail for this PR since they use the current WCPay Dev Tools on `trunk`.

#### Testing instructions

* On your local WCPay clone, checkout the PR branch by running `git fetch && git checkout update/6027-fix-e2e-test-for-wcpay-dev-tools-revamp`
* On your local WCPay clone, reset and setup the e2e tests by running `npm run test:e2e-reset && npm run test:e2e-setup` (you can skip this if you know what you are doing)
* Now, we need to use the upcoming WCPay Dev Tools PR branch for our e2e test; open a terminal at `tests/e2e/deps/wcp-dev-tools`; run the following command `git remote rm origin && git remote add origin git@github.com:Automattic/woocommerce-payments-dev-tools.git && git fetch && git checkout update/reorganize-cleanup-document-dev-tools-options` (this is needed because the e2e plugin clone is done with the `--depth` parameter and is tracking only the `trunk` branch)
* On your local WCPay clone, run the e2e tests with `npm run test:e2e`; they should pass (it will take a couple of many minutes to run the e2e tests).
* To save you time, you can only run the e2e tests that rely on the WCPay Dev Tools:
```
npm run test:e2e -- tests/e2e/specs/upe/
npm run test:e2e -- tests/e2e/specs/upe-split/
npm run test:e2e -- tests/e2e/specs/wcpay/merchant/merchant-payment-settings-manual-capture.spec.js
```

-------------------


**Post merge**
